src/inference/src/dev/core_impl.cpp

```diff
@@ -315,10 +315,7 @@ ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> ov::CoreImpl::compile_mod
             ._cacheManager;
     auto cacheContent = CacheContent{cacheManager};
     if (!forceDisableCache && cacheManager && device_supports_import_export(plugin)) {
-        cacheContent.blobId = CalculateNetworkHash(ov::legacy_convert::convert_model(model, is_new_api()),
-                                                   parsed._deviceName,
-                                                   plugin,
-                                                   parsed._config);
+        cacheContent.blobId = calculate_model_hash(model->clone(), parsed._deviceName, plugin, parsed._config);
```

Comment:
why do we need `clone`? we don't change `ov::Model` inside `calculate_model_hash`

Reply:
Changed clone to const cast inside. we cannot apply pass to constant function